src/openloco/windows/NewsWindow.cpp

```diff
@@ -0,0 +1,1335 @@
+
+            if (news->companyId == company_id::null || companyId == _playerCompany)
+            {
+                soundId = static_cast<audio::sound_id>(_messageSounds[activeMessage]);
+            }
+
+            if (soundId != audio::sound_id::null)
+            {
+                int32_t pan = ui::width() / 2;
+                audio::play_sound(soundId, pan);
+            }
+        }
+
+        if (_word_4F8BE4[activeMessage] & (1 << 1))
+        {
+            _word_525CE0 = 5;
+
+            int16_t y = ui::height() - _word_525CE0;
+
+            if (_gameSpeed != 0 || companyId != 0)
+            {
+                y = ui::height() - news2::windowSize.height;
+                _word_525CE0 = news2::windowSize.height;
+            }
+
+            int16_t x = (ui::width() / 2) - (news2::windowSize.width / 2);
+            gfx::point_t origin = { x, y };
+            uint32_t flags = window_flags::stick_to_front | window_flags::viewport_no_scrolling | window_flags::transparent | window_flags::no_background;
+
+            auto window = WindowManager::createWindow(WindowType::news, origin, news2::windowSize, flags, &news1::events);
+
+            window->widgets = news2::widgets;
+            window->enabled_widgets = common::enabledWidgets;
+
+            common::initEvents();
+
+            window->init_scroll_widgets();
+            window->colours[0] = colour::grey;
+
+            _dword_525CD0 = 0xFFFFFFFF;
+            _dword_525CD4 = 0xFFFFFFFF;
+            _dword_525CD8 = 0xFFFFFFFF;
+            _dword_525CDC = 0xFFFFFFFF;
+
+            news1::initViewport(window);
+        }
+        else
+        {
+            _word_525CE0 = 5;
+
+            int16_t y = ui::height() - _word_525CE0;
+
+            if (_gameSpeed != 0 || companyId != 0)
+            {
+                y = ui::height() - news1::windowSize.height;
+                _word_525CE0 = news1::windowSize.height;
+            }
+
+            int16_t x = (ui::width() / 2) - (news1::windowSize.width / 2);
+            gfx::point_t origin = { x, y };
+            uint32_t flags = window_flags::stick_to_front | window_flags::viewport_no_scrolling | window_flags::transparent;
+
+            auto window = WindowManager::createWindow(WindowType::news, origin, news1::windowSize, flags, &news1::events);
+
+            window->widgets = news1::widgets;
+            window->enabled_widgets = common::enabledWidgets;
+
+            common::initEvents();
+
+            window->init_scroll_widgets();
+            window->colours[0] = colour::translucent(colour::dark_yellow);
+
+            _dword_525CD0 = 0xFFFFFFFF;
+            _dword_525CD4 = 0xFFFFFFFF;
+            _dword_525CD8 = 0xFFFFFFFF;
+            _dword_525CDC = 0xFFFFFFFF;
+
+            news1::initViewport(window);
+        }
+    }
+
+    namespace common
+    {
+        static void initEvents()
+        {
+            ticker::initEvents();
+            news1::initEvents();
```

Comment:
Probably more manageable to put the different windows/styles in a different file, similar to the construction window.

Reply:
I'll wait until I have the construction window working over multiple files before I attempt this. But good idea!